clang/include/clang/ExtractAPI/Serialization/SerializerBase.h

```diff
@@ -28,11 +28,78 @@ struct APISerializerOption {
 };
 
 /// The base interface of serializers for API information.
-class APISerializer {
+template <typename Derived> class APISerializer {
 public:
   /// Serialize the API information to \p os.
   virtual void serialize(raw_ostream &os) = 0;
```

Comment:
It would be nice to turn this one into traverseAPISet and to rename the class to APISetVisitor